WcaOnRails/app/models/competition.rb

```diff
@@ -920,6 +920,25 @@ def set_wcif_events!(wcif_events, current_user)
     reload
   end
 
+  # Takes an array of partial Person WCIF and updates the fields that are not immutable.
+  def update_persons_wcif!(wcif_persons, current_user)
+    persons_schema = { "type" => "array", "items" => User.wcif_json_schema }
+    puts persons_schema.inspect
+    JSON::Validator.validate!(persons_schema, wcif_persons)
+
+    ActiveRecord::Base.transaction do
+      wcif_persons.each do |wcif_person|
+        # registration = registrations.joins(:user).where("users.id = ?", wcif_person["wcaUserId"]).first
+        user = User.includes(:registrations).find(wcif_person["wcaUserId"])
+        registration = user.registrations.find_by_competition_id(self.id)
+        if registration && wcif_person["roles"]
```

Comment:
Could you explain the reasoning behind this? It seems to me that we might want to error out if a registration that the user is trying to update doesn't exist, rather than throwing an error.

If this is the right decision, could you add a comment why this is the right thing to do?

Reply:
Ahh, I think this is definitely worth a comment, then. If we have a registration-less organizer or delegate, we won't be able to give them new roles, which might be desirable.

I think this is yet another reason why we should support registrations for 0 events.

Reply:
Yeah, that's what I stated in PR description =)